pilot/pkg/security/authz/model/util.go

```diff
@@ -27,14 +27,14 @@ import (
 // 4. Suffix match. For example, "*/review" matches "/bookstore/review", "/products/review", etc.
 func stringMatch(a string, list []string) bool {
 	for _, s := range list {
-		if a == s || s == "*" || prefixMatch(a, s) || suffixMatch(a, s) {
+		if a == s || s == "*" || prefixMatch(a, s) || prefixMatch(s, a) || suffixMatch(a, s) || suffixMatch(s, a) {
 			return true
 		}
 	}
 	return false
 }
 
-// prefixMatch checks if string "a" prefix matches "pattern".
+// prefixMatch checks if string "a" suffix matches "pattern".
```

Comment:
Why change the comment?

Reply:
Looks like you have a typo before. Both prefixMatch and suffixMatch have the same comments https://github.com/istio/istio/blob/master/pilot/pkg/security/authz/model/util.go#L46. `prefixMatch` checks if foobar matches the pattern *bar.

Reply:
But this should be changed in the `suffixMatch` function but not here.

Also better to rewrite the comment to: `prefixMatch checks if pattern is a prefix match and if string a has the given prefix`, same for `suffixMatch`.